src/objects/Player.gd

```diff
@@ -49,33 +55,43 @@ func _unhandled_input(event: InputEvent) -> void:
 		camera.rotation.x = clamp(camera.rotation.x - relative.y * MOUSE_SENS.y, -PI / 2, PI / 2)
 		get_tree().set_input_as_handled()
 	elif event.is_action_pressed("shoot"):
-		shoot()
+		if is_active:
+			shoot()
 		get_tree().set_input_as_handled()
 
 
 func _physics_process(delta: float) -> void:
-	var wishdir := Vector2.ZERO
-	var jump_pressed := false
-	if should_control():
-		wishdir = Input.get_vector("move_left", "move_right", "move_backwards", "move_forwards")
-		jump_pressed = Input.is_action_just_pressed("jump")
-
-	var forward_vector := Vector3.FORWARD.rotated(Vector3.UP, rotation.y)
-	var right_vector := Vector3.FORWARD.rotated(Vector3.UP, rotation.y - PI / 2)
-
-	var move_vector := wishdir.x * right_vector + wishdir.y * forward_vector
-
-	velocity.x = move_vector.x * MOVE_SPEED
-	velocity.z = move_vector.z * MOVE_SPEED
-
-	var jumping = false
-
-	if is_on_floor() and jump_pressed:
-		jumping = true
-		velocity.y = JUMP_POWER
-
-	velocity.y -= delta * GRAVITY
-	velocity = move_and_slide_with_snap(velocity, Vector3.ZERO if jumping else Vector3.DOWN, Vector3.UP, true)
+	if iframe_timer > 0.0:
+		iframe_timer -= delta
+		if iframe_timer <= 0.0:
+			is_vulnerable = true
```

Comment:
Minor: This block about iframes should probably be inside the "else" block below